[Term Entry] Dart Type Conversion .toStringAsPrecision() (#4762)
Codecademy/docs: content/dart/concepts/type-conversion/terms/toStringAsPrecision/toStringAsPrecision.md

---
Title: '.toStringAsPrecision()'

> Description: 'Converts this number to a double and returns a string representation of that value.'

Reviewer: ```suggestion
Description: 'Converts a number to a double and returns a string representation with the given precision significant digits.'
```

Subjects:
  - 'Code Foundations'
  - 'Computer Science'
  - 'Mobile Development'
Tags:
  - 'Dart'
  - 'Data Types'
  - 'Methods'
  - 'Numbers'
  - 'Strings'
CatalogContent:
  - 'learn-dart'
  - 'paths/computer-science'
---

> In Dart, **`.toStringAsPrecision()`** is an abstract method used to convert this number to a `double`. This method returns a string representation of that value with exactly precision significant digits.

Reviewer: ```suggestion
In Dart, **`.toStringAsPrecision()`** is an abstract method used to convert this number to a `double`. This method returns a string representation of that value with exact precision significant digits.
```

## Syntax

```pseudo
String toStringAsPrecision(int precision);
```
The parameter `precision` must be an integer satisfying: 1 <= precision <= 21.

## Example

> The following examples illustrates how the `.toStringAsPrecision()` method is utilized to convert this number to a double and returns a string representation of that value:

Reviewer: ```suggestion
The following examples illustrate how the `.toStringAsPrecision()` method is utilized to convert this number to a double and returns a string representation of that value:
```

```dart

> 1.toStringAsPrecision(2);
> 1e15.toStringAsPrecision(3);
> 1234567.toStringAsPrecision(9); 

Reviewer: ```suggestion
void main() {
  String a1 = 1.toStringAsPrecision(2);
  print("Output = " + a1);
  String a2 = 1e15.toStringAsPrecision(3);
  print("Output = " + a2);
  String a3 = 1234567.toStringAsPrecision(9);
  print("Output = " + a3);
}
```

> ```

Reviewer: The reason for this in the code block the code should run just by copying and pasting it
